### PR TITLE
Fix passing mutable options to iOS SDK

### PR DESCRIPTION
**Pull request by marandaneto**
## :loudspeaker: Type of change

- [X] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description
Fix passing mutable options to iOS SDK

## :bulb: Motivation and Context
Not passing `tracesSampleRate` leads to https://github.com/getsentry/sentry-cocoa/issues/1657
Passing a JS `tracesSampler` leads to a crash.

## :pencil: Checklist

- [X] I reviewed submitted code
- [ ] I added tests to verify changes
- [X] All tests passing
- [X] No breaking changes

## :crystal_ball: Next steps
https://github.com/getsentry/sentry-cocoa/issues/1657

### CODE REVIEW

**Review by romtsn — APPROVED** [review]

**Comment by denrase** [comment]
@marandaneto Nice! LGTM!

File: sample/ios/sample.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -387,7 +387,7 @@
 			);
 			inputPaths = (
 				"${PODS_ROOT}/Target Support Files/Pods-sample-sampleTests/Pods-sample-sampleTests-frameworks.sh",
-				"${PODS_XCFRAMEWORKS_BUILD_DIR}/hermes/hermes.framework/hermes",
+				"${PODS_XCFRAMEWORKS_BUILD_DIR}/hermes-engine/hermes.framework/hermes",
```
Comment (marandaneto) [inline sample/ios/sample.xcodeproj/project.pbxproj:390]: @jennmueng did you see this change when migrating to the newer RN engine?
I can't run without this but e2e tests fail now.

**Review by jennmueng — APPROVED** [review]

File: sample/ios/sample.xcodeproj/project.pbxproj (hunk 1 above)
Comment (jennmueng) [reply sample/ios/sample.xcodeproj/project.pbxproj:390]: Works locally, weird.

Comment (marandaneto) [inline sample/ios/sample.xcodeproj/project.pbxproj:390]:
```suggestion
				"${PODS_XCFRAMEWORKS_BUILD_DIR}/hermes/hermes.framework/hermes",
```

File: sample/ios/sample.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -495,7 +495,7 @@
 			);
 			inputPaths = (
 				"${PODS_ROOT}/Target Support Files/Pods-sample/Pods-sample-frameworks.sh",
-				"${PODS_XCFRAMEWORKS_BUILD_DIR}/hermes/hermes.framework/hermes",
+				"${PODS_XCFRAMEWORKS_BUILD_DIR}/hermes-engine/hermes.framework/hermes",
```
Comment (marandaneto) [inline sample/ios/sample.xcodeproj/project.pbxproj:498]:
```suggestion
				"${PODS_XCFRAMEWORKS_BUILD_DIR}/hermes/hermes.framework/hermes",
```

**Comment by marandaneto** [comment]
The failed E2E tests are not related to this PR but rather the flakiness of Appium detecting the buttons of the E2E Test App.

**Pull request merged**